low-level/sensors/impl/centos_7/service_monitor.py

```diff
@@ -0,0 +1,411 @@
+
+            self.services_to_monitor.remove(service)
+
+            return None
+        except DBusException as err:
+            return err
+
+    def check_notactive_services(self):
+        """Raise Alert if any of the not-active services has exceeded
+            the threshould time for inactivity.
+        """
+        temp_not_active_ser = self.not_active_services.copy()
+        for service, start_time in temp_not_active_ser.items():
+            logger.debug(f"{service} : {start_time}, {int(time.time()) - start_time}, {self.max_wait_time}")
+            if int(time.time()) - start_time > self.max_wait_time:
+                state = self.service_status[service]["state"]
+                substate = self.service_status[service]["substate"]
+                pid = self.service_status[service]["pid"]
+                self.not_active_services.pop(service)
+                self.failed_services.append(service)
+                self.raise_alert(service, state, state, substate, substate,
+                                pid , pid, 2)
+
+    def update_status(self, service, state, substate, pid):
+        self.service_status[service] = {
+            "state" : state,
+            "substate" : substate,
+            "pid" : pid
+        }
+
+    def on_prop_changed(self, interface, changed_properties,
+                                invalidated_properties, unit):
+        """Process the `PropertiesChanged` Signal from some service."""
+
+        logger.debug("In on_prop_changed")
+        Iunit = Interface(unit, dbus_interface='org.freedesktop.DBus.Properties')
+        service = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'Id'))
+        state = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'ActiveState'))
+        prev_state = self.service_status[service]["state"]
+
+        logger.debug(f"{service} : {prev_state}:{state}")
+        if prev_state == state:
+            return
+
+        substate = str(Iunit.Get('org.freedesktop.systemd1.Unit', 'SubState'))
+        pid = str(Iunit.Get('org.freedesktop.systemd1.Service', 'ExecMainPID'))
+
+        prev_substate = self.service_status[service]["substate"]
+        prev_pid = self.service_status[service]["pid"]
+
+        logger.info(f"{service} changed state from " + \
+                    f"{prev_state}:{prev_substate} to {state}:{substate}")
+
+        self.update_status(service, state, substate, pid)
+
+        self.action_per_transition(service, prev_state, state,
+                    prev_substate, substate, prev_pid, pid)
+
+
+    def action_per_transition(self, service, prev_state, state,
+                        prev_substate, substate, prev_pid, pid):
+        """Take action according to the state change of the service."""
+        alert_index = -1
+
+        logger.debug(f"in action_per_transition for {service} : " + \
+            f"({prev_state}:{prev_substate}) -> ({state}:{substate})")
+
+        if prev_state == "active" or prev_state == "reloading":
```

Comment:
This whole nested if-else is confusing.

Better to handle one state transition at a time, where ever possible, it would be messy to maintain later as complexity grows. Also per case (if or else), add comment showing state transition being handled, like active->inactive

Reply:
Tried to simplify, please guide if it could be simplified more.